src/drivers/fluid_dsound.c

```diff
@@ -107,8 +107,18 @@ typedef struct
 BOOL CALLBACK
 fluid_dsound_enum_callback(LPGUID guid, LPCTSTR description, LPCTSTR module, LPVOID context)
 {
-    fluid_settings_t *settings = (fluid_settings_t *) context;
-    fluid_settings_add_option(settings, "audio.dsound.device", (const char *)description);
+    fluid_settings_t *settings = (fluid_settings_t *)context;
+    char *name;
+#if _UNICODE
+    int nsz = WideCharToMultiByte(CP_UTF8, 0, description, -1, 0, 0, 0, 0);
+    name = FLUID_ARRAY(char, nsz + 1);
```

Comment:
Is there a reason for allocating +1 byte? From my understanding, when `cchWideChar` is set to -1, the value returned by `WideCharToMultiByte` will always include the NULL byte. So technically, the very last byte of  `name` would be uninitialized. Not sure, if that's intended?

Reply:
This was the result of copy/paste. But you are right: that is what the documentation states, so there should be no need for this precaution.

All drivers fixed.